From 3d7731ed8d36bd50976ffdbae4e1131c630e1557 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 22 Feb 2024 09:13:37 +0100
Subject: [PATCH 0648/1695] dts: stm32: add HSI/CSI calibration on STM32MP15x
 boards

Default enables HSI/CSI calibration on STM32MP15x boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 77910892df9e22cf1db75b6beea4c472f06305a2 dts: stm32: add HSI/CSI calibration on STM32MP15x boards]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I915ba6b74399f21b04925153c3a01aa9a8dc5858
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/286374
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362467
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
---
 core/arch/arm/dts/stm32mp157a-ed1.dts  | 10 ++++++++++
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 10 ++++++++++
 core/arch/arm/dts/stm32mp157d-ed1.dts  | 10 ++++++++++
 core/arch/arm/dts/stm32mp157f-ed1.dts  | 10 ++++++++++
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 10 ++++++++++
 5 files changed, 50 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp157a-ed1.dts b/core/arch/arm/dts/stm32mp157a-ed1.dts
index dcbc035a2..268f81d58 100644
--- a/core/arch/arm/dts/stm32mp157a-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157a-ed1.dts
@@ -215,6 +215,16 @@
 	status = "okay";
 };
 
+&osc_calibration {
+	csi-calibration {
+		status = "okay";
+	};
+
+	hsi-calibration {
+		status = "okay";
+	};
+};
+
 &pwr_regulators {
 	system_suspend_supported_soc_modes = <
 		STM32_PM_CSLEEP_RUN
diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 1daeab5bf..5f6bf0005 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -366,6 +366,16 @@
 	status = "disabled";
 };
 
+&osc_calibration {
+	csi-calibration {
+		status = "okay";
+	};
+
+	hsi-calibration {
+		status = "okay";
+	};
+};
+
 &pwr_regulators {
 	system_suspend_supported_soc_modes = <
 		STM32_PM_CSLEEP_RUN
diff --git a/core/arch/arm/dts/stm32mp157d-ed1.dts b/core/arch/arm/dts/stm32mp157d-ed1.dts
index da905e9f1..e707e6694 100644
--- a/core/arch/arm/dts/stm32mp157d-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157d-ed1.dts
@@ -214,6 +214,16 @@
 	status = "okay";
 };
 
+&osc_calibration {
+	csi-calibration {
+		status = "okay";
+	};
+
+	hsi-calibration {
+		status = "okay";
+	};
+};
+
 &pwr_regulators {
 	system_suspend_supported_soc_modes = <
 		STM32_PM_CSLEEP_RUN
diff --git a/core/arch/arm/dts/stm32mp157f-ed1.dts b/core/arch/arm/dts/stm32mp157f-ed1.dts
index d4fd99a84..062cf1eb8 100644
--- a/core/arch/arm/dts/stm32mp157f-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157f-ed1.dts
@@ -215,6 +215,16 @@
 	};
 };
 
+&osc_calibration {
+	csi-calibration {
+		status = "okay";
+	};
+
+	hsi-calibration {
+		status = "okay";
+	};
+};
+
 &pwr_regulators {
 	system_suspend_supported_soc_modes = <
 		STM32_PM_CSLEEP_RUN
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index d986401c7..7e3113787 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -504,6 +504,16 @@
 	status = "disabled";
 };
 
+&osc_calibration {
+	csi-calibration {
+		status = "okay";
+	};
+
+	hsi-calibration {
+		status = "okay";
+	};
+};
+
 &pwr_regulators {
 	system_suspend_supported_soc_modes = <
 		STM32_PM_CSLEEP_RUN
-- 
2.47.2

